cmd/kubelet/app/server.go

```diff
@@ -1243,6 +1251,55 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	return rl, nil
 }
 
+func parsePreReservedMemoryConfig(config []map[string]string) (map[int]map[v1.ResourceName]uint64, error) {
+	if len(config) == 0 {
+		return nil, nil
+	}
+
+	const (
+		indexKey = "numa-node"
+		typeKkey = "memory-type"
+		limitKey = "limit"
+	)
+
+	keys := []string{indexKey, typeKkey, limitKey}
+
+	// check key first
+	for _, m := range config {
+		for _, key := range keys {
+			if _, exist := m[key]; !exist {
+				return nil, fmt.Errorf("key: %s is missing in given PreReservedMemoryZone flag: %v", key, config)
+			}
+		}
+	}
+
+	parsed := make(map[int]map[v1.ResourceName]uint64, len(config))
+	for _, m := range config {
+		idxInString, _ := m["numa-node"]
+		idx, err := strconv.Atoi(idxInString)
+		if err != nil || idx < 0  {
+			return nil, fmt.Errorf("index conversion error")
+		}
+
+		typeInString, _ := m["memory-type"]
+		v1Type := v1.ResourceName(typeInString)
+		if v1Type != v1.ResourceMemory && !corev1helper.IsHugePageResourceName(v1Type) {
+			return nil, fmt.Errorf("type conversion error")
+		}
+
+		limitInString, _ := m["limit"]
+		limit, err := strconv.Atoi(limitInString)
+		if err != nil || limit < 0  {
+			return nil, fmt.Errorf("limit conversion error")
+		}
```

Comment:
I think that parsing from complex strings is missing here, ex. `1Gi`. Maybe we could use `v1.ParseQuantity()` here